### PR TITLE
Address linter / compiler warnings

### PR DESCRIPTION
**Pull request by elazarg**

### CODE REVIEW

File: src/spec_type_descriptors.hpp [hunk 1]
```diff
@@ -32,13 +32,16 @@ struct EbpfMapDescriptor {
...
 
+constexpr unsigned int DEFAULT_MAP_FD = 0xffffffff;
+
 struct EbpfProgramType {
-    std::string name; // For ease of display, not used by the verifier.
-    const ebpf_context_descriptor_t* context_descriptor {};
-    uint64_t platform_specific_data {}; // E.g., integer program type.
-    std::vector<std::string> section_prefixes;
-    bool is_privileged {};
+    std::string name{}; // For ease of display, not used by the verifier.
```
Comment (elazarg) [inline src/spec_type_descriptors.hpp:38]:
```suggestion
    std::string name{}; /// For ease of display, not used by the verifier.
```

**Comment by dthaler** [comment]
> @dthaler any idea why does the Windows CI fail?

The CICD log says:
> CMake Error at CMakeLists.txt:40 (execute_process):
  execute_process given unknown argument
  "C:/ProgramData/chocolatey/bin/nuget.exe".

Looks like this is somehow due to the changes in CMakeLists.txt

File: CMakeLists.txt [hunk 2]
```diff
@@ -37,6 +37,8 @@ elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "MSVC")
   if (NOT NUGET)
     message("ERROR: You must first install nuget.exe from https://www.nuget.org/downloads")
   else ()
+    # This fails on Windows:
+    # execute_process(${NUGET} ARGS install "Boost" -Version 1.81.0 -ExcludeVersion -OutputDirectory ${CMAKE_BINARY_DIR}/packages)
```
Comment (dthaler) [inline CMakeLists.txt:41]:
```suggestion
    execute_process(COMMAND ${NUGET} ARGS install "Boost" -Version 1.81.0 -ExcludeVersion -OutputDirectory ${CMAKE_BINARY_DIR}/packages)
```

This should do the trick

Comment (dthaler) [reply CMakeLists.txt:41]: I verified the above fix locally.  https://cmake.org/cmake/help/latest/command/execute_process.html explains that COMMAND is required.

**Review by dthaler — APPROVED** [review]

**Pull request merged**